build/Microsoft.DotNet.Cli.Test.targets

```diff
@@ -27,13 +27,28 @@
       <TestTaskEnvironmentVariables Include="TEST_ARTIFACTS=$(TestArtifactsEnv)" />
     </ItemGroup>
 
-    <DotNetTest WorkingDirectory="%(TestProjects.ProjectDir)"
-                EnvironmentVariables="@(TestTaskEnvironmentVariables)"
-                ToolPath="$(Stage2Directory)"
-                Configuration="$(Configuration)"
-                NoBuild="True"
-                NoTrait="category=failing"
-                Xml="$(TestResultXmlDir)/%(TestProjects.OutputName)-testResults.xml" />
+    <PropertyGroup>
+      <RunTestEnvironmentVariables>@(TestTaskEnvironmentVariables)</RunTestEnvironmentVariables>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ProjectsToTest Include="build/test/RunTest.proj">
+        <Properties>
+          CLIBuildDll=$(CLIBuildDll);
+          Configuration=$(Configuration);
+          EnvironmentVariables=$(RunTestEnvironmentVariables);
+          TestProjectName=%(TestProjects.OutputName);
+          TestResultXmlDir=$(TestResultXmlDir);
+          ToolPath=$(Stage2Directory);
+          WorkingDirectory=%(TestProjects.ProjectDir)
+        </Properties>
+      </ProjectsToTest>
+    </ItemGroup>
+
+    <MSBuild
```

Comment:
look at the batched <MSBuild target just below.

They way that one works is that invokes the default target [or specified] on the project that it points to. Each batch has the option to override Properties inside of that proj. What these are doing are creating one Item for each test project and defining the appropriate properties to make it run. They then invoke this new test project which does the actual invocation of dotnet-test. By batching on an MSBuild task we're telling MSBuild it's ok to treat each of these projects as independent, which allows it to run them on multiple cores!